src/controllers/assistant.js

```diff
@@ -45,91 +46,169 @@ function AssistantController(context) {
     }
 
-    let assistantConfiguration;
-    if (!context.env.ASSISTANT_CONFIGURATION) {
-      throw new ErrorWithStatusCode('The Assistant Configuration is not defined.', STATUS.SYS_ERROR);
+    const { authInfo: { profile: apikeyProfile } } = attributes;
+
+    const callerImsOrgId = apikeyProfile?.getImsOrgId() ?? headerImsOrgId;
+    if (!callerImsOrgId) {
+      throw new ErrorWithStatusCode(
+        'Invalid request: A valid ims-org-id is not associated with your api-key.',
+        STATUS.UNAUTHORIZED,
+      );
     }
-    try {
-      assistantConfiguration = JSON.parse(context.env.ASSISTANT_CONFIGURATION);
-    } catch (error) {
-      throw new ErrorWithStatusCode(`Could not parse the Assistant Configuration: ${error.message}`, STATUS.SYS_ERROR);
+  }
+
+  /**
+   * Process and validate the request context.
+   * @param requestContext
+   * @param requestContext.data Object containing the request parameters
+   * @param requestContext.data.command String indicating the command to execute
+   * @param requestContext.data.prompt String containing the user prompt
+   * @param requestContext.data.options Object containing the request options
+   * @param requestContext.data.options.htmlContent String containing the HTML content
+   * @param requestContext.data.options.selector CSS selectors of the block being processed
+   * @param requestContext.data.options.imageUrl String base64 encoded image
+   * @param requestContext.attributes Object containing request processing (api-key, profile, etc.)
+   * @param requestContext.pathInfo Object containing the request path and headers
+   * @param requestContext.pathInfo.headers Object containing the request headers
+   * @param requestContext.pathInfo.headers.x-api-key String api-key
+   * @param requestContext.pathInfo.headers.x-gw-ims-org-id String ims-org-id
+   * @param requestContext.env Object containing the environment variables (secrets, etc.)
+   * @param requestContext.env.ASSISTANT_CONFIGURATION Object containing the assistant configuration
+   * @param requestContext.env.ASSISTANT_PROMPTS Object containing the assistant prompts
+   * @returns the validated and parsed request context data
+   */
+  function parseRequestContext(requestContext) {
+    // Validate input data.
+    if (!requestContext) {
+      throw new ErrorWithStatusCode(
+        'Invalid request: missing request context.',
+        STATUS.BAD_REQUEST,
+      );
     }
+    if (!isObject(requestContext)
+      || !requestContext.data
+      || !requestContext.attributes
+      || !requestContext.pathInfo
+    ) {
+      throw new ErrorWithStatusCode(
+        'Invalid request: invalid request context format.',
+        STATUS.BAD_REQUEST,
+      );
+    }
+
+    const { data, pathInfo: { headers }, attributes } = requestContext;
+
+    validateRequestData(data);
+    validateRequestAttributes(requestContext.attributes, headers['x-gw-ims-org-id']);
+    const contextEnv = processAndValidateContextEnv(context.env);
```

Comment:
Since the request is a little complicated, I decided to validate part individually, instead of in a long stream, hopefully making it easier to comprehend.